docs/generate_extension_categories.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+#!/usr/bin/env bash
+
+set -e
+
+output_file_name="$1"/api/factory_categories.csv
+
+echo Writing into $output_file_name
+
+echo "\"Category Name\", \"Source File\"" > $output_file_name
+
+find . -name *.h -exec grep -HF "std::string category()" {} \; | while IFS= read -r line
+do
+  tokens=( $line )
+  category=${tokens[6]}
```

Comment:
I think it would be cleaner if each file had a single line `const char* FOO_CATEGORY = "foo's category";`. That way you could do a single regex on one line to obtain this. Should be fairly robust as these are short strings, so shouldn't span lines.

Reply:
done